core/src/test/java/org/elasticsearch/index/query/FuzzyQueryBuilderTest.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.query;
+
+import org.apache.lucene.index.Term;
+import org.apache.lucene.search.FuzzyQuery;
+import org.apache.lucene.search.MultiTermQuery;
+import org.apache.lucene.search.Query;
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.ParseFieldMatcher;
+import org.elasticsearch.common.lucene.BytesRefs;
+import org.elasticsearch.common.unit.Fuzziness;
+import org.elasticsearch.index.mapper.MappedFieldType;
+import org.elasticsearch.index.query.support.QueryParsers;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.junit.Test;
+
+import java.io.IOException;
+
+import static org.hamcrest.Matchers.is;
+
+public class FuzzyQueryBuilderTest extends BaseQueryTestCase<FuzzyQueryBuilder> {
+
+    @Override
+    protected FuzzyQueryBuilder doCreateTestQueryBuilder() {
+        // unmapped fields
+        String fieldName = randomAsciiOfLengthBetween(1, 10);
+        Object value = randomAsciiOfLengthBetween(1, 50);
+
+        // mapped fields
+        if (randomBoolean()) {
+            fieldName = randomFrom(mappedFieldNames);
+            switch(fieldName) {
+                case DATE_FIELD_NAME:
+                    value = new DateTime(System.currentTimeMillis(), DateTimeZone.UTC);
+                    break;
+                case INT_FIELD_NAME:
+                    value = randomIntBetween(0, 10);
+                    break;
+                case DOUBLE_FIELD_NAME:
+                    value = randomDouble() * 10;
+                    break;
+                case BOOLEAN_FIELD_NAME:
+                    value = randomBoolean();
+                    break;
+            } // OBJECT_FIELD_NAME and STRING_FIELD_NAME assigned to random string
+        }
```

Comment:
I left this comment before but it didn't get addressed yet. I think it would be important to test unmapped fields against other types besides String. Thats what BaseTermQueryTestCase does in doCreateTestQueryBuilder. I think we should do exactly the same here.